apis/apps/v1alpha1/apicast_types.go

```diff
@@ -28,46 +28,81 @@ import (
 
+	// Number of replicas of the APIcast Deployment.
 	// +optional
 	Replicas *int64 `json:"replicas,omitempty"`
+	// Secret reference to a Kubernetes Secret containing the admin portal
+	// endpoint URL. The Secret must be located in the same namespace.
 	// +optional
 	AdminPortalCredentialsRef *v1.LocalObjectReference `json:"adminPortalCredentialsRef,omitempty"`
+	// Secret reference to a Kubernetes secret containing the gateway
+	// configuration. The Secret must be located in the same namespace.
 	// +optional
 	EmbeddedConfigurationSecretRef *v1.LocalObjectReference `json:"embeddedConfigurationSecretRef,omitempty"`
+	// Kubernetes Service Account name to be used for the APIcast Deployment. The
+	// Service Account must exist beforehand.
 	// +optional
 	ServiceAccount *string `json:"serviceAccount,omitempty"`
+	// Image allows overriding the default APIcast gateway container image.
+	// This setting should only be used for dev/testing purposes. Setting
+	// this disables automated upgrades of the image.
 	// +optional
 	Image *string `json:"image,omitempty"`
+	// ExposedHost is the domain name used for external access. By default no
+	// external access is configured.
 	// +optional
 	ExposedHost *APIcastExposedHost `json:"exposedHost,omitempty"`
+	// DeploymentEnvironment is the environment for which the configuration will
+	// be downloaded from 3scale (Staging or Production), when using APIcast.
+	// The value will also be used in the header X-3scale-User-Agent in the
+	// authorize/report requests made to 3scale Service Management API. It is
+	// used by 3scale for statistics.
 	// +optional
 	DeploymentEnvironment *DeploymentEnvironmentType `json:"deploymentEnvironment,omitempty"` // THREESCALE_DEPLOYMENT_ENV
+	// DNSResolverAddress can be used to specify a custom DNS resolver address
+	// to be used by OpenResty.
 	// +optional
 	DNSResolverAddress *string `json:"dnsResolverAddress,omitempty"` // RESOLVER
+	// EnabledServices can be used to specify a list of service IDs used to
+	// filter the configured services.
 	// +optional
 	EnabledServices []string `json:"enabledServices,omitempty"` // APICAST_SERVICES_LIST
+	// ConfigurationLoadMode can be used to set APIcast's configuration load mode.
 	// +optional
 	// +kubebuilder:validation:Enum=boot;lazy
 	ConfigurationLoadMode *string `json:"configurationLoadMode,omitempty"` // APICAST_CONFIGURATION_LOADER
+	// LogLevel controls the log level of APIcast's OpenResty logs.
 	// +optional
 	// +kubebuilder:validation:Enum=debug;info;notice;warn;error;crit;alert;emerg
 	LogLevel *string `json:"logLevel,omitempty"` // APICAST_LOG_LEVEL
+	// PathRoutingEnabled can be used to enable APIcast's path-based routing
+	// in addition to to the default host-based routing.
 	// +optional
 	PathRoutingEnabled *bool `json:"pathRoutingEnabled,omitempty"` // APICAST_PATH_ROUTING
+	// ResponseCodesIncluded can be set to log the response codes of the responses
```

Comment:
```suggestion
	// ResponseCodesIncluded when set to true, stores the response codes that the managed API returns
```

Reply:
I'm not sure about this one. It does not store anything right? My understanding was that it just includes the response codes as part of the log :thinking:

Reply:
https://github.com/3scale/APIcast/blob/master/doc/parameters.md#apicast_response_codes

Reply:
It basically starts counting response codes per application and then reports to 3scale backend (apisonator). The that data is stored (by backend listener) in redis and available in the dashboard

Reply:
I think the description that's currently shown is good enough. No need to go into that detail about redis storage

Reply:
I agree. My comment was to let you both know what this parameter activates.